core/arch/riscv64/riscv64.asm

```diff
@@ -208,16 +208,52 @@ ADDRTAKEN_LABEL(safe_read_asm_recover:)
  */
         DECLARE_EXPORTED_FUNC(dr_try_start)
 GLOBAL_LABEL(dr_try_start:)
-        addi      ARG1, ARG1, TRY_CXT_SETJMP_OFFS
+        addi     ARG1, ARG1, TRY_CXT_SETJMP_OFFS
         j        GLOBAL_REF(dr_setjmp)
         END_FUNC(dr_try_start)
 
-/*
+/* We save only callee-saved registers and ra: ra, SP, x8/fp, x9, x18-x27, f8-9, f18-27:
+ * a total of 26 reg_t (64-bit) slots. See definition of dr_jmp_buf_t.
+ *
  * int dr_setjmp(dr_jmp_buf_t *buf);
  */
         DECLARE_FUNC(dr_setjmp)
 GLOBAL_LABEL(dr_setjmp:)
-/* FIXME i#3544: Not implemented */
+        sd       ra, 0 (ARG1)
+        sd       t0, ARG_SZ (ARG1)
+        mv       t0, sp
```

Comment:
I think this `mv t0, sp` should be before the above `sd t0, ...`

Reply:
done

Reply:
Fixed https://github.com/DynamoRIO/dynamorio/pull/6071/commits/1451c1cbcbfbef45d68fa62cb0c32144f99f686c